src/accordion/AccordionState.ts

```diff
@@ -1,117 +1,73 @@
-  /**
-   * The current selected(controlled) accordion's `id`.
-   */
-  selectedIds: (string | null)[];
-  /**
-   * Whether the accodion selection should be manual.
-   *
-   * @default true
-   */
-  manual: boolean;
-  /**
-   * Allow to open multiple accordion items
-   *
-   * @default false
-   */
-  allowMultiple: boolean;
-  /**
-   * Allow to toggle accordion items
-   *
-   * @default false
-   */
-  allowToggle: boolean;
-  /**
-   * Lists all the panels.
-   */
-  panels: CompositeState["items"];
-};
-
-export type AccordionActions = CompositeActions & {
-  /**
-   * Moves into and selects an accordion by its `id`.
-   */
-  select: CompositeActions["move"];
-  /**
-   * Moves into and unSelects an accordion by its `id` if it's already selected.
-   */
-  unSelect: CompositeActions["move"];
-  /**
-   * Sets `selectedId`.
-   */
-  setSelectedId: React.Dispatch<React.SetStateAction<string | null>>;
-  /**
-   * Sets `selectedIds`.
-   */
-  setSelectedIds: React.Dispatch<React.SetStateAction<(string | null)[]>>;
-  /**
-   * Registers a accordion panel.
-   */
-  registerPanel: CompositeActions["registerItem"];
-  /**
-   * Unregisters a accordion panel.
-   */
-  unregisterPanel: CompositeActions["unregisterItem"];
-};
-
-export type AccordionInitialState = CompositeInitialState &
-  Pick<
-    Partial<AccordionState>,
-    "selectedId" | "selectedIds" | "manual" | "allowMultiple" | "allowToggle"
-  > & {
-    /**
-     * Set default selected id(uncontrolled)
-     *
-     * @default null
-     */
-    defaultSelectedId?: string | null;
-    /**
-     * Handler that is called when the selectedId changes.
-     */
-    onSelectedIdChange?: (value: string | null) => void;
-    /**
-     * Set default selected ids(uncontrolled)
-     *
-     * @default []
-     */
-    defaultSelectedIds?: (string | null)[];
-    /**
-     * Handler that is called when the selectedIds changes.
-     */
-    onSelectedIdsChange?: (value: (string | null)[]) => void;
-  };
+export function useAccordionState(
+  props: Partial<AccordionInitialStateSingle>,
+): SingleOverloadReturn;
+
+export function useAccordionState(
+  props: Partial<AccordionInitialStateMulti>,
+): MultiOverloadReturn;
+
+export function useAccordionState(
+  props: Partial<AccordionInitialState>,
+): SingleOverloadReturn;
 
-export type AccordionStateReturn = AccordionState & AccordionActions;
+export function useAccordionState(
+  props: Partial<AccordionInitialState>,
+): MultiOverloadReturn;
 
 export function useAccordionState(
-  props: AccordionInitialState = {},
+  props: Partial<AccordionInitialState> = {},
 ): AccordionStateReturn {
   const {
-    selectedId: selectedIdProp,
-    defaultSelectedId = null,
-    onSelectedIdChange,
-    selectedIds: selectedIdsProp,
-    defaultSelectedIds = [],
-    onSelectedIdsChange,
-    allowMultiple = false,
-    allowToggle: allowToggleProp = false,
     manual = true,
+    allowToggle: allowToggleProp = false,
+    allowMultiple,
```

Comment:
Not needed !

Reply:
we need to do this then since it needed to be fallback to false

![image](https://user-images.githubusercontent.com/35374649/98941363-7d7e4b80-2512-11eb-97b9-7a77302445dc.png)

Reply:
I am just saying we can remove line 43, since we are using props.allowMultiiple everywhere.